x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -0,0 +1,246 @@
+  return {
+    id: htmlIdGenerator()(),
+    label: `Anomalies count: ${jobId}`,
+    joins: [
+      {
+        // Left join is the id from the type of field (e.g. world_countries)
+        leftField: field,
+        right: {
+          id: 'anomaly_count',
+          type: SOURCE_TYPES.TABLE_SOURCE,
+          __rows: getAnomalyRows(anomalies, jobId),
+          __columns: [
+            {
+              name: 'entityValue',
+              type: 'string',
+            },
+            {
+              name: 'count',
+              type: 'number',
+            },
+            {
+              name: 'max_severity',
+              type: 'number',
+            },
+          ],
+          // Right join/term is the field in the doc you’re trying to join it to (foreign key - e.g. US)
+          term: 'entityValue',
+        },
+      },
+    ],
+    sourceDescriptor: {
+      type: 'EMS_FILE',
+      id: layerId,
+    },
+    style: {
+      type: 'VECTOR',
+      // @ts-ignore missing style properties. Remove once 'VectorLayerDescriptor' type is updated
+      properties: {
+        icon: { type: STYLE_TYPE.STATIC, options: { value: 'marker' } },
+        fillColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: {
+            color: 'Blue to Red',
+            colorCategory: 'palette_0',
+            fieldMetaOptions: { isEnabled: true, sigma: 3 },
+            type: COLOR_MAP_TYPE.ORDINAL,
+            field: {
+              name: 'count',
+              origin: FIELD_ORIGIN.JOIN,
+            },
+            useCustomColorRamp: false,
+          },
+        },
+        lineColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: { color: '#3d3d3d', fieldMetaOptions: { isEnabled: true } },
+        },
+        lineWidth: { type: STYLE_TYPE.STATIC, options: { size: 1 } },
+      },
+      isTimeAware: true,
+    },
+    visible,
+    type: 'VECTOR',
+  };
+};
+
+interface Props {
+  anomalies: AnomaliesTableRecord[];
+  jobIds: string[];
+}
+
+interface MLEMSTermJoinConfig extends EMSTermJoinConfig {
+  jobId: string;
+}
+
+export const AnomaliesMap: FC<Props> = ({ anomalies, jobIds }) => {
+  const [layerList, setLayerList] = useState<VectorLayerDescriptor[]>([]);
+  const [EMSSuggestions, setEMSSuggestions] = useState<
+    Array<MLEMSTermJoinConfig | null> | undefined
+  >();
+  const {
+    services: { maps: mapsPlugin },
+  } = useMlKibana();
+
+  const getEMSTermSuggestions = useCallback(async (): Promise<void> => {
+    if (mapsPlugin) {
+      const suggestions = await Promise.all(
+        jobIds.map(async (jobId) => {
+          const entityValues = new Set<string>();
+          let entityName;
+          for (let i = 0; i < anomalies.length; i++) {
+            if (jobId === anomalies[i].jobId && anomalies[i].entityValue !== '') {
+              entityValues.add(anomalies[i].entityValue);
+
+              if (!entityName) {
+                entityName = anomalies[i].entityName;
+              }
+            }
```

Comment:
question: is it possible that some entity values are not related to the entity name eventually?

Reply:
Shouldn't be https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/server/models/results_service/build_anomaly_table_items.js#L21